source/validate_id.cpp

```diff
@@ -1612,17 +1612,202 @@ bool idUsage::isValid<OpCompositeConstruct>(
     const spv_instruction_t *inst, const spv_opcode_desc opcodeEntry) {}
 #endif
 
-#if 0
 template <>
-bool idUsage::isValid<OpCompositeExtract>(
-    const spv_instruction_t *inst, const spv_opcode_desc opcodeEntry) {}
-#endif
+bool idUsage::isValid<SpvOpCompositeExtract>(const spv_instruction_t* inst,
+                                             const spv_opcode_desc) {
+  std::string instr_name =
```

Comment:
This does a bunch of allocations and a (slow for now) lookup.  But the common case is there is no error, so it's wasteful.
I'd rather make a lambda here that you then call when needed.